HolyshiftService/AuthManager.cs

```
@@ -0,0 +1,132 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+
+public class RequestDto
+{
+    public string Email { get; set; }
+    public string Password { get; set; }
+}
+
+public class ResponseDto
+{
+    public string AccessToken { get; set; }
+    public string RefreshToken { get; set; }
+    public string Message { get; set; }
+}
+public class RegistereRequestDto
+{
+    public string Email { get; set; }
+    public string Password { get; set; }
+    public string UserName { get; set; } 
+}
+
+public class RegisterResponseDto
+{
+    public string Message { get; set; }
+}
+
+public class AuthManager
+{
+    private readonly HolyShiftDbContext _db;
+    private readonly AuthService _jwtService;
+    private readonly PasswordHashService _passwordHashService;
+    private readonly IUserDao _userDao;
+
+    public AuthManager(HolyShiftDbContext db, AuthService jwtService, PasswordHashService passwordHashService, IUserDao userDao)
+    {
+        _db = db;
+        _jwtService = jwtService;
+        _passwordHashService = passwordHashService;
+        _userDao = userDao;
+    }
+
+    public async Task<ResponseDto> SignIn(RequestDto requestDto)
+    {
+        var responseDto = new ResponseDto();
+
+        if (string.IsNullOrEmpty(requestDto.Email) || string.IsNullOrEmpty(requestDto.Password))
+        {
+            responseDto.Message = "Invalid request body";
+            return responseDto;
```

For now let's just return an error code and let the front end decide on the exact message. Because localization on the frontend is slightly easier?

---

I did that first but shouldnt we know were is the problem ? (if its a wrong email or password or for ex in signup Email already exists)